app/assets/javascripts/initializers/initializeReadingListIcons.js

```diff
@@ -33,14 +33,17 @@ function highlightButton(button) {
 }
 
 function addReadingListCountToHomePage() {
-  var user = userData();
-  var readingListCount;
-  if (user && document.getElementById('reading-list-count')) {
-    readingListCount =
-      user.reading_list_ids.length > 0 ? user.reading_list_ids.length : '';
-    document.getElementById('reading-list-count').innerHTML = readingListCount;
-    document.getElementById('reading-list-count').dataset.count =
-      user.reading_list_ids.length;
+  const user = userData();
+  const readingListContainers = document.querySelectorAll(
+    '.reading-list-count',
+  );
+  if (user && readingListContainers) {
+    readingListContainers.forEach(function (e) {
+      let readingListCount =
```

Comment:
```suggestion
      const readingListCount =
```
(just a suggestion) - I don't think we ever reassign this?